### PR TITLE
Update conventions section

### LINKED ISSUE #165: Update conventions section
Mention mypy and Pydocstyle.

### PR DESCRIPTION
**Pull request by NoB0**
Add mypy, docformatter, and pydocstyle in Conventions section.

Fixes #165

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +82,8 @@ If you want to specify a specific commit as the source of the package append the
 
 ## Conventions
 
-We follow the [IAI Python Style Guide](https://github.com/iai-group/styleguide/tree/main/python).
+We follow the [IAI Python Style Guide](https://github.com/iai-group/styleguide/tree/main/python), except that we assume Python 3.9.
```
Comment (kbalog) [inline README.md:85]: Those guidelines will also be updated to Python 3.9 (https://github.com/iai-group/guidelines/issues/45), so you can skip that part. But it'd be good to have a separate Prerequisites section where we state the Python version and any other dependencies.

File: README.md [hunk 2]
```diff
@@ -82,7 +82,8 @@ If you want to specify a specific commit as the source of the package append the
...
+We have additional pre-commit hooks for [mypy](https://mypy.readthedocs.io/en/stable/), [docformatter](https://docformatter.readthedocs.io/en/latest/), and [pydocstyle](http://www.pydocstyle.org/en/stable/index.html) (see [Wiki](https://github.com/iai-group/DialogueKit/wiki/Pre-commit) for more details).
```
Comment (kbalog) [inline README.md:86]: mypy and docformatter will be part of the IAI Guidelines (https://github.com/iai-group/guidelines/issues/44), so you can skip those, only mention pydocstyle (and only if we decide not to use it everywhere, cf. https://github.com/iai-group/guidelines/issues/48).

**Review by kbalog — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (NoB0) [reply README.md:85]: Done.

**Comment by NoB0** [comment]
Mypy, docformatter, and pydocstyle will be added to the IAI conventions.

**Pull request closed without merging**